src/viewer.js

```diff
@@ -320,32 +398,245 @@ function _coordinateFormatScoord3d2Geometry(coordinates, pyramid) {
+    nameCodedConceptMeaning
+  );
+
+  const index = evaluations.findIndex((e) =>
+    isContentItemsEqual(e, evaluation)
+  );
+
+  if (index > -1) {
+    evaluations[index] = evaluation;
+  } else {
+    evaluations.push(evaluation);
+  }
+
+  feature.set(Enums.InternalProperties.Evaluations, evaluations);
+  console.debug(`Evaluations of feature (${feature.getId()}):`, evaluations);
+}
+
+/**
+ * Updates feature measurement properties
+ *
+ * @param {object} map The map instance
+ * @param {object} feature The feature instance
+ * @returns {void}
+ */
+const _updateFeatureMeasurementProperties = (map, feature) => {
+  /**
+   * Open Layers side-effect: Geometry will be changed
+   * inside the formCircle call which causes errors
+   * if const variable is used for geometry
+   */
+  let geometry = feature.getGeometry().clone();
+  if (geometry instanceof LineStringGeometry) {
+    feature.set(Enums.FeatureMeasurement.Length, getLength(geometry));
+  } else if (geometry instanceof CircleGeometry) {
+    geometry = fromCircle(geometry);
+    feature.set(Enums.FeatureMeasurement.Area, getArea(geometry));
+  } else if (geometry instanceof PolygonGeometry) {
+    feature.set(Enums.FeatureMeasurement.Area, getArea(geometry));
+  }
+};
+
+/**
+ * Generate feature measurements from its measurement properties
+ *
+ * @param {object} map The map instance
+ * @param {object} feature The feature instance
+ * @returns {void}
  */
-function _getROIFromFeature(feature, pyramid) {
-  if (feature !== undefined && feature !== null) {
-    const geometry = feature.getGeometry();
-    const scoord3d = _geometry2Scoord3d(geometry, pyramid);
-    const properties = feature.getProperties();
-    // Remove geometry from properties mapping
-    const geometryName = feature.getGeometryName();
-    delete properties[geometryName];
-    const uid = feature.getId();
-    return new ROI({ scoord3d, properties, uid });
-  }
-  return
+function _updateFeatureMeasurements(map, feature) {
+  const measurements = feature.get(Enums.InternalProperties.Measurements) || [];
+  const area = feature.get(Enums.FeatureMeasurement.Area);
+  const length = feature.get(Enums.FeatureMeasurement.Length);
+
+  if (!area && !length) return;
+
+  const unitSuffixToMeaningMap = {
+    μm: "micrometer",
+    mm: "millimeter",
+    m: "meters",
+    km: "kilometers",
+  };
+
+  let measurement;
+  const view = map.getView();
+  const unitSuffix = getUnitSuffix(view);
+  const unitCodedConceptValue = unitSuffix;
+  const unitCodedConceptMeaning = unitSuffixToMeaningMap[unitSuffix];
+
+  if (area) {
+    const nameCodedConceptValue = "Area";
+    const nameCodedConceptMeaning = "42798000";
+    measurement = getMeasurementContentItem(
+      area,
+      nameCodedConceptValue,
+      nameCodedConceptMeaning,
+      unitCodedConceptValue,
+      unitCodedConceptMeaning
+    );
+  }
+
+  if (length) {
+    const nameCodedConceptValue = "Length";
+    const nameCodedConceptMeaning = "410668003";
```

Comment:
Value and Meaning are switched:

```suggestion
    const nameCodedConceptValue = "410668003";
    const nameCodedConceptMeaning = "Length";
```

Reply:
Fixed on my local env, not sure why it was switched here, maybe old version?